### ISSUE TITLE
react-music-player is unmaintaned

### ISSUE THREAD

**Issue by mradermaxlol** [navidrome/navidrome#1651]
### Description

I think we have a problem.
![image](https://user-images.githubusercontent.com/11138966/158056900-e2c1656b-6b7e-4723-a6db-e2004baf9370.png)

Time to change the default player or fork the existing one, I guess?

IIRC there's a number of issues that are "blocked by upstream", the webplayer in particular. Forking or changing the used player in its entirety could be a way to get rid of those long-standing issues, e.g. lack of gapless playback.

**Comment by J4gQBqqR** [comment]
Important features, such as gapless playback and replaygain, have been on hold because of react backend not supporting them. Another reason to consider changing it.

**Comment by gildesmarais** [comment]
Since the initial post contained no link, here's the repo of the dependency: https://github.com/lijinke666/react-music-player

A quick [search for a useful fork](https://useful-forks.github.io/?repo=lijinke666/react-music-player) returned nothing which could act as a quality replacement.

Since some issues seem to be blocked by this player anyway, it might be a good time to get rid of it. Has anybody experience with building a player, i.e. with Audio API, state of the art libraries, etc?

How would a way forward look like here?

**Comment by deluan** [comment]
Currently Navidrome is using its own fork: https://github.com/navidrome/react-music-player with the "Play Next" bug fixed.

The plan is to build our own player, Here are some thoughts about it. The main discussion about this is here: https://discord.com/channels/671335427726114836/704303730660737113/861006516315881522

The main requirements for a new player are:
- Written in React/Material-UI, so it can easily blend into ND's UI and be themable
- Compatible with most/all browsers and main media formats. For this it should be based on HTML5 Media Player. Most recent players out there are.
- Gapless Playback (super important!)
- Normalization (Replay Gain support)
- Control remote API (for jukebox)
- Easy to extend the UI (to add more options)
- It may be a separated project

So my original idea is to create a react player, don't need to be too fancy, implemented with [useAudioPlayer](https://github.com/E-Kuerschner/useAudioPlayer), that allows you to specify an audio provider. 

This way we can have a html5 audio provider for playback in the browser, and a remote audio provider, for controlling the jukebox. useAudioPlayer may pose a problem as we are not sure it offers gapless playback. If not, we may get around it by creating a new provider for it, that wraps the [Stitches library](https://github.com/sudara/stitches)

**Comment by gildesmarais** [comment]
Wow, thank you for the detailed response and updating the labels. :)

Looks like the collection of requirements for this important component already started. ;) I'll continue reading in the Discord and see how to continue from there.

**Comment by GeneralChauhan** [comment]
Is it still open??

**Comment by deluan** [comment]
> Is it still open??

Discussions about a new player moved to here: https://github.com/navidrome/navidrome/discussions/2285

Work on it has not started yet.
